pkg/core/core.go

```diff
@@ -36,137 +35,81 @@ type ModelRegistryService struct {
 }
 
 // NewModelRegistryService creates a new instance of the ModelRegistryService, initializing it with the provided gRPC client connection.
-// It sets up the necessary context and artifact types in the MLMD service.
+// It _assumes_ the necessary MLMD's Context, Artifact, Execution types etc. are already setup in the underlying MLMD service.
 //
 // Parameters:
 //   - cc: A gRPC client connection to the underlying MLMD service
 func NewModelRegistryService(cc grpc.ClientConnInterface) (api.ModelRegistryApi, error) {
+	typesMap, err := GetTypeIdsIntoTypesMap(cc)
+	if err != nil { // early return in case type Ids cannot be retrieved
+		return nil, err
+	}
 
 	client := proto.NewMetadataStoreServiceClient(cc)
 
-	// Setup the needed Type instances if not existing already
+	return &ModelRegistryService{
+		mlmdClient:  client,
+		typesMap:    typesMap,
+		openapiConv: &generated.OpenAPIConverterImpl{},
+		mapper:      mapper.NewMapper(typesMap),
+	}, nil
+}
 
-	registeredModelReq := proto.PutContextTypeRequest{
-		CanAddFields: canAddFields,
-		ContextType: &proto.ContextType{
-			Name: registeredModelTypeName,
-			Properties: map[string]proto.PropertyType{
-				"description": proto.PropertyType_STRING,
-				"state":       proto.PropertyType_STRING,
-			},
-		},
-	}
+func GetTypeIdsIntoTypesMap(cc grpc.ClientConnInterface) (map[string]int64, error) {
```

Comment:
Nit: `into` usually indicates a transformation. I assume you meant to `get type ids in the types map` here, but the meaning is not the same. To avoid the `the` article, I'd rewrite it as

```suggestion
func GetTypeIdsFromTypesMap(cc grpc.ClientConnInterface) (map[string]int64, error) {
```

Reply:
The semantic is not of `From` tho?
`FromTypesMap` looks wrong, I'm not procuding something _from a TypesMap_ ?

Reply:
See https://github.com/opendatahub-io/model-registry/pull/267/commits/c8e1b26262897ccf5062f45e61455f84def4b2b1 if that solves your nit request please.
I'd really like to move fwd with this PR as it is needed from MC

Reply:
Thanks for addressing it properly while I was away. Regarding the confusion, the new name is definitely clearer, well done :)